unison-src/transcripts/alias-many.output.md

```diff
@@ -18,174 +18,251 @@ Let's try it!
-  167. bug : a -> b
-  168. todo : a -> b
+  1.   AlreadyExists : IOError
+  2.   builtin type Boolean
+  3.   Boolean.not : Boolean -> Boolean
+  4.   type BufferMode
+  5.   BufferMode.BlockBuffering : BufferMode
+  6.   BufferMode.LineBuffering : BufferMode
+  7.   BufferMode.NoBuffering : BufferMode
+  8.   BufferMode.SizedBlockBuffering : Nat -> BufferMode
+  9.   builtin type Bytes
+  10.  Bytes.++ : Bytes -> Bytes -> Bytes
+  11.  Bytes.at : Nat -> Bytes -> Optional Nat
+  12.  Bytes.drop : Nat -> Bytes -> Bytes
+  13.  Bytes.empty : Bytes
+  14.  Bytes.flatten : Bytes -> Bytes
+  15.  Bytes.fromList : [Nat] -> Bytes
+  16.  Bytes.size : Bytes -> Nat
+  17.  Bytes.take : Nat -> Bytes -> Bytes
+  18.  Bytes.toList : Bytes -> [Nat]
+  19.  builtin type Char
+  20.  Char.fromNat : Nat -> Char
+  21.  Char.toNat : Char -> Nat
+  22.  Debug.watch : Text -> a -> a
+  23.  unique type Doc
+  24.  Doc.Blob : Text -> Doc
+  25.  Doc.Evaluate : Term -> Doc
+  26.  Doc.Join : [Doc] -> Doc
+  27.  Doc.Link : Link -> Doc
+  28.  Doc.Signature : Term -> Doc
+  29.  Doc.Source : Link -> Doc
+  30.  EOF : IOError
+  31.  type Either a b
+  32.  Either.Left : a -> Either a b
+  33.  Either.Right : b -> Either a b
+  34.  type FileMode
+  35.  FileMode.Append : FileMode
+  36.  FileMode.Read : FileMode
+  37.  FileMode.ReadWrite : FileMode
+  38.  FileMode.Write : FileMode
+  39.  builtin type Float
+  40.  Float.* : Float -> Float -> Float
+  41.  Float.+ : Float -> Float -> Float
+  42.  Float.- : Float -> Float -> Float
+  43.  Float./ : Float -> Float -> Float
+  44.  Float.abs : Float -> Float
+  45.  Float.acos : Float -> Float
+  46.  Float.acosh : Float -> Float
+  47.  Float.asin : Float -> Float
+  48.  Float.asinh : Float -> Float
+  49.  Float.atan : Float -> Float
+  50.  Float.atan2 : Float -> Float -> Float
+  51.  Float.atanh : Float -> Float
+  52.  Float.ceiling : Float -> Int
+  53.  Float.cos : Float -> Float
+  54.  Float.cosh : Float -> Float
+  55.  Float.eq : Float -> Float -> Boolean
+  56.  Float.exp : Float -> Float
+  57.  Float.floor : Float -> Int
+  58.  Float.fromText : Text -> Optional Float
+  59.  Float.gt : Float -> Float -> Boolean
+  60.  Float.gteq : Float -> Float -> Boolean
+  61.  Float.log : Float -> Float
+  62.  Float.logBase : Float -> Float -> Float
+  63.  Float.lt : Float -> Float -> Boolean
+  64.  Float.lteq : Float -> Float -> Boolean
+  65.  Float.max : Float -> Float -> Float
+  66.  Float.min : Float -> Float -> Float
+  67.  Float.pow : Float -> Float -> Float
+  68.  Float.round : Float -> Int
+  69.  Float.sin : Float -> Float
+  70.  Float.sinh : Float -> Float
+  71.  Float.sqrt : Float -> Float
+  72.  Float.tan : Float -> Float
+  73.  Float.tanh : Float -> Float
+  74.  Float.toText : Float -> Text
+  75.  Float.truncate : Float -> Int
+  76.  IO.clientSocket : Text
+                         -> Text
+                         ->{##IO} Either IOError ##Socket
```

Comment:
This transcript calls `builtins.merge` at the start, which looks like it includes names for the builtin `IO` functions but not for the `IO` type itself? (since it is showing up as `##IO` here). That's a little odd.

We could use `builtins.mergeio` to give names for both the `IO` type and all these functions. 

Or just have `builtins.merge` include the `IO` type and functions.

@aryairani WDYT?

Reply:
Everything related to the new `IO` stuff is now under the `io2` namespace when you merge it, I believe. I made the declared types go there, too, for instance. I presume just giving them that name in the hand-built declaration works, since there's no renaming DSL like there is for the builtin types.